## ***Pima Indians Diabetes Exploratory Analysis & Data Preprocessing***
---

### *About Dataset*
#### *Context*
This dataset [source](https://www.kaggle.com/datasets/uciml/pima-indians-diabetes-database/data?select=diabetes.csv) is originally from the National Institute of Diabetes and Digestive and Kidney Diseases. The objective of the dataset is to diagnostically predict whether or not a patient has diabetes, based on certain diagnostic measurements included in the dataset. Several constraints were placed on the selection of these instances from a larger database. In particular, all patients here are females at least 21 years old of Pima Indian heritage.

#### *Content*
The datasets consists of several medical predictor variables and one target variable, Outcome. Predictor variables includes the number of pregnancies the patient has had, their BMI, insulin level, age, and so on.

#### *Acknowledgements*
Smith, J.W., Everhart, J.E., Dickson, W.C., Knowler, W.C., & Johannes, R.S. (1988). Using the ADAP learning algorithm to forecast the onset of diabetes mellitus. In Proceedings of the Symposium on Computer Applications and Medical Care (pp. 261--265). IEEE Computer Society Press.

### Who is Pima Indians ?

"The Pima (or Akimel O'odham, also spelled Akimel O'otham, "River People", formerly known as Pima) are a group of Native Americans living in an area consisting of what is now central and southern Arizona. The majority population of the surviving two bands of the Akimel O'odham are based in two reservations: the Keli Akimel O'otham on the Gila River Indian Community (GRIC) and the On'k Akimel O'odham on the Salt River Pima-Maricopa Indian Community (SRPMIC)." Wikipedia

---
---

In [16]:
import pandas as pd
# import seaborn as sns/
import matplotlib.pyplot as plt
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt


In [5]:
dia=pd.read_csv("diabetes.csv")
dia.head()

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
0            6      148             72             35        0  33.6   
1            1       85             66             29        0  26.6   
2            8      183             64              0        0  23.3   
3            1       89             66             23       94  28.1   
4            0      137             40             35      168  43.1   

   DiabetesPedigreeFunction  Age  Outcome  
0                     0.627   50        1  
1                     0.351   31        0  
2                     0.672   32        1  
3                     0.167   21        0  
4                     2.288   33        1

In [8]:
## Shape of the dataset

dia.shape

(768, 9)

- The dataset has 768 observations(rows) and 9 variables(columns)


| Variables | Description | Type | Nature |
|-----------|-------------|------|--------|
|Pregnancies|Number of pregnancies|Quantitative|Discrete|
|Glucose    |Plasma glucose concentration|Quantitative|Continuous|
|BloodPressure|Diastolic blood pressure (mm Hg)|Quantitative|Continuous|
|SkinThickness|Triceps skin fold thickness (mm)|Quantitative|Continuous|
|Insulin|2-hour serum insulin(muU/ml)|Quantitative|Continuous|
|BMI|Body mass index|Quantitative|Continuous|
|DiabetesPedigreeFunction|Diabetes pediggree function|Quantitative|Continuous|
|Age|Age(years)|Quantitative|Discrete|
|Outcome|Diabetes status (0=No,1=Yes)|Qualitative|Discrete(Binary)|

B. Exploratory Data Analysis

In [13]:
quant_vars = dia.drop(columns=['Outcome'])
desc = pd.DataFrame({
    "Min": quant_vars.min(),
    "Max": quant_vars.max(),
    "Range": quant_vars.max() - quant_vars.min(),
    "Mean": quant_vars.mean(),
    "Median": quant_vars.median(),
    "Mode": quant_vars.mode().iloc[0],
    "Variance": quant_vars.var(),
    "Std_Dev": quant_vars.std(),
    
})
desc

Min     Max    Range        Mean    Median  \
Pregnancies                0.000   17.00   17.000    3.845052    3.0000   
Glucose                    0.000  199.00  199.000  120.894531  117.0000   
BloodPressure              0.000  122.00  122.000   69.105469   72.0000   
SkinThickness              0.000   99.00   99.000   20.536458   23.0000   
Insulin                    0.000  846.00  846.000   79.799479   30.5000   
BMI                        0.000   67.10   67.100   31.992578   32.0000   
DiabetesPedigreeFunction   0.078    2.42    2.342    0.471876    0.3725   
Age                       21.000   81.00   60.000   33.240885   29.0000   

                            Mode      Variance     Std_Dev  
Pregnancies                1.000     11.354056    3.369578  
Glucose                   99.000   1022.248314   31.972618  
BloodPressure             70.000    374.647271   19.355807  
SkinThickness              0.000    254.473245   15.952218  
Insulin                    0.000  13281.180078  115.244002  
BMI                       32.000     62.159984    7.884160  
DiabetesPedigreeFunction   0.254      0.109779    0.331329  
Age                       22.000    138.303046   11.760232